From: jvrao <jvrao@linux.vnet.ibm.com>
Subject: [PATCH 077/111] fs/9p: Remove the redundant rsize calculation in v9fs_file_write()
Git-commit: 8d40fa2492eb3dcf02468eef2f6bba450be42b22
Patch-Mainline: 2.6.37
References: FATE#311639

    the same calculation is done in p9_client_write

	upstream commit 8d40fa2492eb3dcf02468eef2f6bba450be42b22

    Signed-off-by: Venkateswararao Jujjuri <jvrao@linux.vnet.ibm.com>
    Signed-off-by: Badari Pulavarty <pbadari@us.ibm.com>
    Signed-off-by: Eric Van Hensbergen <ericvh@gmail.com>
Signed-off-by: Harsh Prateek Bora <harsh@linux.vnet.ibm.com>
Acked-by: Bruce Rogers <brogers@suse.com>
---
 fs/9p/vfs_file.c |   10 ++--------
 1 files changed, 2 insertions(+), 8 deletions(-)

diff --git a/fs/9p/vfs_file.c b/fs/9p/vfs_file.c
index 613935e..48fbd60 100644
--- a/fs/9p/vfs_file.c
+++ b/fs/9p/vfs_file.c
@@ -221,7 +221,7 @@ v9fs_file_write(struct file *filp, const char __user * data,
 {
 	ssize_t retval;
 	size_t total = 0;
-	int n, rsize;
+	int n;
 	struct p9_fid *fid;
 	struct p9_client *clnt;
 	struct inode *inode = filp->f_path.dentry->d_inode;
@@ -234,8 +234,6 @@ v9fs_file_write(struct file *filp, const char __user * data,
 	fid = filp->private_data;
 	clnt = fid->clnt;
 
-	rsize = fid->iounit ? fid->iounit : clnt->msize - P9_IOHDRSZ;
-
 	retval = generic_write_checks(filp, &origin, &count, 0);
 	if (retval)
 		goto out;
@@ -248,11 +246,7 @@ v9fs_file_write(struct file *filp, const char __user * data,
 		goto out;
 
 	do {
-		if (count < rsize)
-			rsize = count;
-
-		n = p9_client_write(fid, NULL, data+total, origin+total,
-									rsize);
+		n = p9_client_write(fid, NULL, data+total, origin+total, count);
 		if (n <= 0)
 			break;
 		count -= n;
-- 
1.7.1.1

